src/main/scala/loamstream/googlecloud/HailSupport.scala

```diff
@@ -61,7 +61,7 @@ object HailSupport {
     def hail(args: Any*)(implicit scriptContext: LoamScriptContext): LoamCmdTool = {
       
       require(
-          scriptContext.executionEnvironment.isGoogle, 
+          scriptContext.settings.isGoogle, 
```

Comment:
I'm sure I've asked this before in other PRs, but asking again to solidify the reason in my mind...

Why have `.isType` methods, which need to be maintained and such over just using `.isInstanceOf[T]`? I seem to remember there being a reason for it.

Reply:
A big reason is cultural preference, born out of collective experience: Most Java and Scala devs consider `isInstanceOf` a smell that should be avoided if possible, in favor of doing something type-safe.

Here, I'd consider the actual run-time class of `scriptContext.settings` an irrelvant detail, at least from the point of view of `HailSupport`, or basically anything not in `Settings.scala`.  Saying "I don't care what this thing is, precisely, just tell me if it's a Google thing or not" is a useful bit of abstraction that keeps code that doesn't need to know `settings`'s runtime type from being coupled to `settings`'s runtime type.